### ISSUE TITLE
WSGI middleware breaks when handling badly encoded URLs

### ISSUE THREAD

**Issue by wosc** [bugsnag/bugsnag-python#185]
### Description

In public-facing applications, we cannot control what kinds of URLs will be requested by clients -- which means we may have to deal with wrong or broken encoding.

### Issue

`bugsnag.wsgi.middleware.add_wsgi_request_data_to_notification` uses `webob.Request.path_url`, which raises an exception if the URL path is encoded improperly. It would not break if it instead e.g. used the already existing helper function `bugsnag.wsgi.request_path`.

### Environment

Library versions:

- python version: both 2.7 and 3.7
- bugsnag-python version: 3.6.0

### Example code snippet

```python
import unittest
import unittest.mock
import urllib.parse

class PathEncodingTest(unittest.TestCase):

    environ = {
        # https://github.com/python/cpython/blob/master/Lib/wsgiref/simple_server.py#L85
        'PATH_INFO': urllib.parse.unquote('/%83', 'latin-1'),
        'SCRIPT_NAME': '',
        'SERVER_NAME': 'localhost',
        'SERVER_PORT': '80',
        'wsgi.url_scheme': 'http',
    }

    def test_wrongly_encoded_url_should_not_raise(self):
        import bugsnag.wsgi.middleware
        notification = unittest.mock.Mock()
        notification.request_config.wsgi_environ = self.environ
        # This currently raises UnicodeDecodeError
        bugsnag.wsgi.middleware.add_wsgi_request_data_to_notification(
            notification)
        # but I'd expect something like this:
        self.assertEqual(
            'http://localhost/%83',
            notification.add_tab.call_args_list[0][0][1]['url'])

    def test_using_the_already_existing_helper_function_works_somewhat(self):
        import bugsnag.wsgi
        import webob
        r = webob.Request(self.environ)
        self.assertEqual(
            # Not sure how correct this is, but at least it doesn't break
            'http://localhost/%C2%83',
            r.application_url + bugsnag.wsgi.request_path(self.environ))

    def test_werkzeug_instead_of_webob_also_handles_it_without_error(self):
        import werkzeug
        r = werkzeug.Request(self.environ)
        self.assertEqual('http://localhost/%83', r.base_url)

if __name__ == '__main__':
    unittest.main()
```

<details><summary>Error messages:</summary>

```
..E
======================================================================
ERROR: test_wrongly_encoded_url_should_not_raise (__main__.EncodingTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "../foo.py", line 21, in test_wrongly_encoded_url_should_not_raise
    notification)
  File "/Users/schnerring/Downloads/venv/lib/python3.7/site-packages/bugsnag/wsgi/middleware.py", line 28, in add_wsgi_request_data_to_notification
    "url": request.path_url,
  File "/Users/schnerring/Downloads/venv/lib/python3.7/site-packages/webob/request.py", line 467, in path_url
    bpath_info = bytes_(self.path_info, self.url_encoding)
  File "/Users/schnerring/Downloads/venv/lib/python3.7/site-packages/webob/descriptors.py", line 70, in fget
    return req.encget(key, encattr=encattr)
  File "/Users/schnerring/Downloads/venv/lib/python3.7/site-packages/webob/request.py", line 165, in encget
    return bytes_(val, 'latin-1').decode(encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x83 in position 1: invalid start byte

----------------------------------------------------------------------
Ran 3 tests in 0.099s

FAILED (errors=1)
```
</details>

**Comment by mattdyoung** [comment]
Hi @wosc 

Thanks for the report. We'll look into how soon we can schedule a fix for this.

**Comment by kattrali** [comment]
Fixed in [3.6.1](https://github.com/bugsnag/bugsnag-python/releases/tag/v3.6.1)
